From 448c387b8274c4416378e51cda29d135613c4c22 Mon Sep 17 00:00:00 2001
From: Raphael Kubo da Costa <raphael.kubo.da.costa@intel.com>
Date: Wed, 13 Feb 2019 10:17:50 +0100
Subject: IceTransportWithTransportChannel: Initialize |thread_checker_| in
 declaration
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

This works around https://gcc.gnu.org/bugzilla/show_bug.cgi?id=89305, which
causes GCC to fail to build the code due to |thread_checker_| being const
there and not having a declared constructor.

    ../../api/ice_transport_factory.cc: In constructor ‘webrtc::{anonymous}::IceTransportWithTransportChannel::IceTransportWithTransportChannel(std::unique_ptr<cricket::IceTransportInternal>)’:
    ../../api/ice_transport_factory.cc:31:3: error: uninitialized const member in ‘const class rtc::ThreadChecker’ [-fpermissive]
       IceTransportWithTransportChannel(
       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    ../../api/ice_transport_factory.cc:45:28: note: ‘const rtc::ThreadChecker webrtc::{anonymous}::IceTransportWithTransportChannel::thread_checker_’ should be initialized
       const rtc::ThreadChecker thread_checker_;
                                ^~~~~~~~~~~~~~~

Bug: chromium:819294
Change-Id: I750e8cdd796b3b0e076de01194cf7de988ac4ce2
Reviewed-on: https://webrtc-review.googlesource.com/c/122820
Reviewed-by: Karl Wiberg <kwiberg@webrtc.org>
Commit-Queue: Raphael Kubo da Costa (rakuco) <raphael.kubo.da.costa@intel.com>
Cr-Commit-Position: refs/heads/master@{#26662}

diff --git a/api/ice_transport_factory.cc b/api/ice_transport_factory.cc
index b632f09133..6cc399aef7 100644
--- a/api/ice_transport_factory.cc
+++ b/api/ice_transport_factory.cc
@@ -42,7 +42,7 @@ class IceTransportWithTransportChannel : public IceTransportInterface {
   }
 
  private:
-  const rtc::ThreadChecker thread_checker_;
+  const rtc::ThreadChecker thread_checker_{};
   const std::unique_ptr<cricket::IceTransportInternal> internal_
       RTC_GUARDED_BY(thread_checker_);
 };
-- 
2.17.1

